Adding AWS provisioner example (#1817)
aws/karpenter: website/content/en/preview/AWS/provisioning.md

  Learn AWS-specific Karpenter provisioning settings
---

Provisioner settings specific to Karpenter for the AWS cloud provider are described here.
The following example shows optional and required settings for a Karpenter provisioner for AWS:

```yaml
apiVersion: karpenter.sh/v1alpha5
kind: Provisioner
metadata:
  name: default
spec:
  requirements:                      

>     - key: karpenter.sh/capacity-type         # optional, set to on-demand by default

Reviewer: This isn't really cloud provider specific, we may want to leave it off entirely for this part since it's explained in the provisioner API docs: https://karpenter.sh/v0.10.0/provisioner/#:~:text=%2D%20key%3A%20%22karpenter.sh/capacity%2Dtype%22%20%23%20If%20not%20included%2C%20the%20webhook%20for%20the%20AWS%20cloud%20provider%20will%20default%20to%20on%2Ddemand

      operator: In

>       values: ["spot"]

Reviewer: I'm not sure if this is the right place to put this but a question that might come up based on the default being on-demand is: "what happens when both are in the list of values" which in that case, we'd default to spot.

  limits:   
    resources:
      cpu: 1000                               # optional, recommended to limit total cluster CPU
      memory: 1000Gi                          # optional, recommended to limit total cluster memory

>   provider:

Reviewer: we need to document all the parameters in this struct https://github.com/aws/karpenter/blob/main/pkg/cloudprovider/aws/apis/v1alpha1/provider.go#L29

Author: I added all the parameters in that struct to the example. All are described below. Let me know if AMIFamily, InstanceProfile, and Tags examples are okay.

  Learn AWS-specific Karpenter provisioning settings
---

Provisioner settings specific to Karpenter for the AWS cloud provider are described here.
The following example shows optional and required settings for a Karpenter provisioner for AWS:

```yaml
apiVersion: karpenter.sh/v1alpha5
kind: Provisioner
metadata:
  name: default
spec:
  requirements:                      
    - key: karpenter.sh/capacity-type         # optional, set to on-demand by default, spot if both are listed
      operator: In
      values: ["spot"]
  limits:   
    resources:

>       cpu: 1000                               # optional, recommended to limit total cluster CPU
>       memory: 1000Gi                          # optional, recommended to limit total cluster memory

Reviewer: ```suggestion
      cpu: 1000                               # optional, recommended to limit total provisioned CPUs
      memory: 1000Gi                          # optional, recommended to limit total provisioned memory
```

  provider:
    subnetSelector:                           # required
      karpenter.sh/discovery: ${CLUSTER_NAME}

>     securityGroupSelector:                    # required

Reviewer: this is not required when using the launchTemplate field.

Maybe we can note it:

`# required when not using [launchTemplate](#LaunchTemplate)`

Author: Okay. Done.

>     securityGroupSelector:                    # required when not using [launchTemplate](../launch-template.md)

Reviewer: code block, so can't link :)

```suggestion
    securityGroupSelector:                    # required when not using launchTemplate
```

Author: Ah, of course. Fixed.

      karpenter.sh/discovery: ${CLUSTER_NAME}

>     instanceProfile: MyInstanceProfile        # optional, if already set on controller Getting Started

Reviewer: ```suggestion
    instanceProfile: MyInstanceProfile        # optional, if already set in controller args
```

>     AMIFamily: AmazonLinux2                   # optional
>     InstanceProfile: MyInstanceProfile        # optional, set in [Getting Started]({{<ref "../getting-started/getting-started-with-eksctl/" >}}), but can overrride here
>     Tags:                                     # optional, override default tags or set your own

Reviewer: ```suggestion
    amiFamily: AL2                   # optional, AL2 by default
    instanceProfile: MyInstanceProfile        # optional, set in [Getting Started]({{<ref "../getting-started/getting-started-with-eksctl/" >}}), but can overrride here
    tags:                                     # optional, override default tags or set your own
```

Author: Done.

>       MySpecialTag: true

Reviewer: It's probably better to use an actual `string` here rather than a `bool` where the type is ambiguous. I'm not sure what a good value is here, so I just put `foo`.  

```suggestion
      MySpecialTag: foo
```

Author: Done

```

> ### SecurityGroupSelector (required)

Reviewer: same here on the `required` caveat

Additionally, include a resource requirement in the workload manifest. This will cause the GPU dependent pod will be scheduled onto the appropriate node.

> Here is an example of an accelerator resource in workload a manifest (e.g., pod):

Reviewer: ```suggestion
Here is an example of an accelerator resource in a workload manifest (e.g., pod):
```
